models/repository/repository.go

```diff
@@ -15,18 +23,44 @@ type RepositoryStruct struct {
 	RepositoryID int64
 	Owner        sql.NullString
 	Name         sql.NullString
+	WebhookKey   string
 	database     db.DB
 }
 
-func NewRepository(id int64, repositoryID int64, owner string, name string) *RepositoryStruct {
+func GenerateWebhookKey(seed string) string {
+	h := md5.New()
+	io.WriteString(h, strconv.FormatInt(time.Now().Unix(), 10))
+	io.WriteString(h, seed)
+	token := fmt.Sprintf("%x", h.Sum(nil))
+
+	return token
+}
+
+func NewRepository(id int64, repositoryID int64, owner string, name string, webhookKey string) *RepositoryStruct {
 	if repositoryID <= 0 {
 		return nil
 	}
-	repository := &RepositoryStruct{ID: id, RepositoryID: repositoryID, Owner: sql.NullString{String: owner, Valid: true}, Name: sql.NullString{String: name, Valid: true}}
+	repository := &RepositoryStruct{ID: id, RepositoryID: repositoryID, Owner: sql.NullString{String: owner, Valid: true}, Name: sql.NullString{String: name, Valid: true}, WebhookKey: webhookKey}
 	repository.Initialize()
 	return repository
 }
 
+func FindRepositoryByRepositoryID(repositoryID int64) *RepositoryStruct {
```

Comment:
__[golint]__
* [golint] exported function FindRepositoryByRepositoryID should have comment or be unexported